src/runtime/object.cpp

```diff
@@ -1133,6 +1133,55 @@ extern "C" LEAN_EXPORT void lean_extract_mpz_value(lean_object * o, mpz_t v) {
 }
 #endif
 
+extern "C" LEAN_EXPORT obj_res lean_mpz_of_usize(usize n) {
+    return alloc_mpz(mpz::of_size_t(n));
+}
+
+extern "C" LEAN_EXPORT obj_res lean_mpz_of_uint64(uint64 n) {
+    return alloc_mpz(mpz(n));
+}
+
+extern "C" LEAN_EXPORT obj_res lean_mpz_of_uint32(uint32 n) {
+    static_assert(sizeof(uint32_t) <= sizeof(unsigned), "unsigned should be at least the size of a uint32");
+    return alloc_mpz(mpz(static_cast<unsigned>(n)));
+}
+
+extern "C" LEAN_EXPORT obj_res lean_mpz_of_unsigned(unsigned n) {
+    return alloc_mpz(mpz(n));
+}
+
+extern "C" LEAN_EXPORT obj_res lean_mpz_of_isize(size_t a) {
+    return alloc_mpz(mpz(static_cast<int64>(static_cast<ptrdiff_t>(a))));
+}
+
+extern "C" LEAN_EXPORT obj_res lean_mpz_of_int64(uint64_t n) {
+    return alloc_mpz(mpz(static_cast<int64>(n)));
+}
+
+extern "C" LEAN_EXPORT obj_res lean_mpz_of_int32(uint32_t n) {
+    static_assert(sizeof(uint32_t) <= sizeof(int), "int should be at least the size of a uint32");
+    return alloc_mpz(mpz(static_cast<int>(static_cast<int32_t>(n))));
```

Comment:
Why the pair of casts vs letting a constructor overload handle it?

Reply:
However, perhaps adding more constructor overloads to `mpz` would be clearer?